src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -387,6 +394,11 @@ namespace cryptonote
     m_fluffy_blocks_enabled = !get_arg(vm, arg_no_fluffy_blocks);
     m_offline = get_arg(vm, arg_offline);
     m_disable_dns_checkpoints = get_arg(vm, arg_disable_dns_checkpoints);
+    m_sync_pruned_blocks = !command_line::get_arg(vm, cryptonote::arg_no_sync_pruned_blocks) 
+        && command_line::get_arg(vm, cryptonote::arg_fast_block_sync) != 0
+        && command_line::get_arg(vm, cryptonote::arg_prune_blockchain);
```

Comment:
This could probably be simplified. The extra checks are [likely unnecessary](https://github.com/monero-project/monero/pull/5915#issuecomment-531772037), but I figured it's best to double check.

Reply:
I'm now fully sure that these checks are redundant. Since they may cause weird edge cases, they were removed.